packages/react/src/useTable.tsx

```diff
@@ -53,78 +53,42 @@ export const useTable = <
     }
   }, []);
 
-  const fieldSelectionMap = useMemo(() => {
-    if (options?.select) return options.select;
-
-    if (!options?.columns) return undefined; // Use the default selection
-
-    const selectionMap: FieldSelection = {};
-    if (!metadata) return selectionMap;
-
-    for (const column of options.columns) {
-      if (typeof column === "string") {
-        selectionMap[column] = true;
-        continue;
-      }
-
-      if (isRelatedFieldColumn(column)) {
-        // We need to find the related model field from the metadata
-        // to determine whether the field is a has-one or has-many relationship,
-        // and build the correct selection map for it.
-        const fieldMetadata = metadata.fields.find((field) => field.apiIdentifier === column.field);
-        if (!fieldMetadata) {
-          throw new Error(`Field '${column.field}' not found in metadata`);
-        }
-
-        if (
-          fieldMetadata.configuration.__typename === "GadgetHasOneConfig" ||
-          fieldMetadata.configuration.__typename === "GadgetBelongsToConfig"
-        ) {
-          selectionMap[column.field] = {
-            [column.relatedField]: true,
-          };
-        } else {
-          selectionMap[column.field] = {
-            edges: {
-              node: {
-                [column.relatedField]: true,
-              },
-            },
-          };
-        }
-      }
+  const useTableDataGenerator = useMemo(() => {
```

Comment:
We might have to dig into the details here, but I'm not sure that a class is the best way to extract these helpers -- any reason not to extract the individual helpers as functions? 

The reason I think a class might be a challenge down the road is that it introduces additional state management that might be a bit tricky to get right in terms of memoization, etc.

Reply:
I used a class instead of individual helper functions because I personally find it annoying to pass in the same parameters, such as `metadata` and `columns,` to all these functions all over the place.